File: functions/dictionary.lua
--Convert any dictionary that has 255 or less characters
local function asctodec(str)
    local decarr = {}
    for i = 1, #str do
        local char = str:sub(i, i)
        table.insert(decarr, string.byte(char))
    end
    return decarr
end

-- Function to convert decimal array back to ASCII string
local function dectoasc(decarr)
    local temp = {}
    for _, value in ipairs(decarr) do
        table.insert(temp, string.char(value))
    end
    return table.concat(temp)
end

return {asctodec = asctodec, dectoasc = dectoasc}

File: cpp/dictionary.cpp
#include "dictionary.h"
#include <vector>
#include <string>

// Convert ASCII string to decimal array
std::vector<int> Dictionary::asctodec(const std::string& str) {
    std::vector<int> decarr;
    for (char c : str) {
        decarr.push_back(static_cast<int>(c));
    }
    return decarr;
}

// Convert decimal array back to ASCII string
std::string Dictionary::dectoasc(const std::vector<int>& decarr) {
    std::string result;
    for (int value : decarr) {
        result += static_cast<char>(value);
    }
    return result;
}
